### PR TITLE
Feat/landlord dashboard

### PR DESCRIPTION
**Pull request by baudoin-csp**
## UX/UI

The user experience has been fully aligned with the Figma design specifications for consistency. (See the Figma design and the current implementation)

<table>
  <tr>
    <td>figma</td>
    <td>current</td>
  </tr>
  <tr>
    <td style="text-align: center;">Figma Design</td>
    <td style="text-align: center;">Current Implementation</td>
  </tr>
</table>

I created the `IssueStackLandlord` to manage navigation from the dashboard to both the `IssueDetails` and `Messaging` screens.

The functionality to list the residence was implemented using a carousel (to display images horizontally using a slider).

carousel

The navigation flow has been clearly defined to ensure seamless transitions between screens, and the dashboard can be accessed through the home section in the menu.

navigation

Messaging is currently missing, but Mahdi will implement it soon as all prerequisites are in place.  
Errors encountered when fetching a residence are now correctly handled, with a retry button provided for users to attempt fetching data again (see the video below). You can also notice that the case where a landlord has no residence is covered.

[See video](https://github.com/user-attachments/assets/48be529d-acca-497f-b4d3-147d28c665c2)

## Conclusion

This update improves the `LandlordDashboard` with better navigation, error handling, and a polished design that matches the Figma specs. Messaging is still pending but ready to be implemented. These changes make the dashboard more reliable and user-friendly.

### CODE REVIEW

File: app/Navigators/IssueStackLandlord.tsx
Comment (Rcc999) [inline app/Navigators/IssueStackLandlord.tsx:1]: The IssueStackLandlord is nicely structured, using createStackNavigator to set up navigation in a clean and maintainable way.

File: app/Navigators/LandlordDrawerNavigator.tsx
Comment (Rcc999) [inline app/Navigators/LandlordDrawerNavigator.tsx:1]: The inclusion of IssueStackLandlord in LandlordDrawerNavigator is a good practice that ensures modularity and keeps navigation logic clear. It integrates well into the broader navigation framework, maintaining consistency with the app's design.

File: app/__tests__/LandloardDashboard.test.tsx
Comment (Rcc999) [inline app/__tests__/LandloardDashboard.test.tsx:1]: The tests for LandlordDashboard are incredibly thorough, covering a wide range of scenarios such as loading states, error handling, and edge cases like no residences or maintenance requests.

While the tests cover a lot of ground, they are quite dense and could benefit from more comments explaining the purpose of specific test cases or setups. For example, adding a brief note about why mockRejectedValueOnce is used in certain scenarios would make the code easier to follow for others.

File: types/types.ts
Comment (Rcc999) [inline types/types.ts:1]: The addition of the LandlordStackParamList type is a great improvement, as it ensures type safety across the navigation stack for the landlord-related screens. Defining explicit parameters like requestID for IssueDetails reflects good attention to detail and helps prevent runtime errors.

File: app/screens/landlord/LandlordDashboard.tsx
Comment (Rcc999) [inline app/screens/landlord/LandlordDashboard.tsx:1]: The LandlordDashboard component is well-structured and effectively utilizes React hooks like useEffect, useState, and useMemo for managing state and performance optimizations. The clear separation of concerns between data fetching, maintenance issue calculations, and UI rendering ensures the component is easy to read and maintain. The inclusion of test IDs for nearly all elements is a great addition for testing and debugging.

While the component handles loading, error, and data states effectively, the fetchResidence function is doing too much by fetching landlords, residences, apartments, and maintenance requests all in one go. Breaking it into smaller functions or delegating some of the logic to helper utilities would improve readability and maintainability.

**Review by Rcc999 — APPROVED** [review]
This pull request does an excellent job of improving the app's navigation and user experience with the addition of the IssueStackLandlord and LandlordDashboard components. The LandlordDashboard is thoughtfully designed, with robust state handling for various scenarios like loading, error, and data display. The addition of type safety with LandlordStackParamList and the comprehensive test suite further strengthens the reliability of these features.

However, there’s room to optimize the data fetching logic in the LandlordDashboard for better readability and separation of concerns. Despite this, the overall implementation is solid, and the changes are well-aligned with the app’s goals. Great job on this PR—it’s ready to be merged! ✅

File: app/screens/landlord/LandlordDashboard.tsx
Comment (baudoin-csp) [reply app/screens/landlord/LandlordDashboard.tsx:1]: I think it's too complicated to separate all the logic from fetchResidence function as the calls are asynchronous and depend on each other.

File: types/types.ts
Comment (baudoin-csp) [reply types/types.ts:1]: Thanks!

File: app/__tests__/LandloardDashboard.test.tsx
Comment (baudoin-csp) [reply app/__tests__/LandloardDashboard.test.tsx:1]: I added comments :)

**Pull request merged**